app/src/main/java/com/nativo/sampleapp/ViewAdapter/GridViewAdapter.kt

```diff
@@ -76,12 +88,7 @@ class GridViewAdapter(context: Context, private val gridView: GridView) :
             articleTitle?.setText(R.string.sample_title)
 
             view.setOnClickListener {
-                context.startActivity(
-                    Intent(
-                        Intent.ACTION_VIEW,
-                        Uri.parse(CLICK_OUT_URL)
-                    )
-                )
+                _itemClickListener?.invoke(item)
             }
 
             if (shouldShowPlaceAd(item)) {
```

Comment:
What I would recommend is have an 'else' condition after this, so if 'shouldShowPlaceAd()' is false, then you add a click listener.